app/components/common/Navigation.vue

```diff
@@ -0,0 +1,534 @@
+<script>
+  const CODECOMBAT = 'codecombat'
+  const CODECOMBAT_CHINA = 'koudashijie'
+  const OZARIA = 'ozaria'
+  const OZARIA_CHINA = 'aojiarui'
+
+  // Use in local development to inform which repo you are in
+  const LOCALHOST_FAKE_HOST = () => application.isProduction() ? null : CODECOMBAT // 'codecombat' | 'ozaria'
+
+  /**
+   * Unified navigation bar component between CodeCombat and Ozaria.
+   * This must be copied exactly to the Ozaria repo.
+   */
+  export default Vue.extend({
+    computed: {
+      isOldBrowser () {
+        if (window.features.china && $.browser) {
+          if (!($.browser.webkit || $.browser.mozilla || $.browser.msedge)) {
+            return true
+          }
+          const majorVersion = $.browser.versionNumber
+          if ($.browser.mozilla && majorVersion < 25) {
+            return true
+          }
+          if ($.browser.chrome && majorVersion < 72) { // forbid some chinese browser
+            return true
+          }
+          if ($.browser.safari && majorVersion < 6) { // 6 might have problems with Aether, or maybe just old minors of 6: https://errorception.com/projects/51a79585ee207206390002a2/errors/547a202e1ead63ba4e4ac9fd
+            return true
+          }
+        }
+        return false
+      },
+
+      isCodeCombat () {
+        if (LOCALHOST_FAKE_HOST() === CODECOMBAT) {
+          return true
+        }
+        return document.location.host.includes(CODECOMBAT) || document.location.host.includes(CODECOMBAT_CHINA)
+      },
+
+      isOzaria () {
+        if (LOCALHOST_FAKE_HOST() === OZARIA) {
+          return true
+        }
+        return document.location.host.includes(OZARIA) || document.location.host.includes(OZARIA_CHINA)
+      },
+
+      cocoBaseURL () {
+        if (this.isCodeCombat) {
+          return ''
+        }
+
+        if (!application.isProduction()) {
+          return `${document.location.protocol}//codecombat.com`
+        }
+
+        // We are on ozaria domain.
+        return `${document.location.protocol}//${document.location.host}`
+          .replace(OZARIA, CODECOMBAT)
+          .replace(OZARIA_CHINA, CODECOMBAT_CHINA)
+      },
+
+      ozBaseURL () {
+        if (this.isOzaria) {
+          return ''
+        }
+
+        if (!application.isProduction()) {
+          return `${document.location.protocol}//ozaria.com`
+        }
+
+        // We are on codecombat domain.
+        return `${document.location.protocol}//${document.location.host}`
+          .replace(CODECOMBAT, OZARIA)
+          .replace(CODECOMBAT_CHINA, OZARIA_CHINA)
+      }
+    },
+
+    created () {
+      // Bind the global values to the vue component.
+      this.me = me
```

Comment:
Silly Vue question: is setting `this.me` something we can do generally in all our components, rather than manually in each one? Or, does this interact with the version of `me` that we (sometimes?) put in the store?

Reply:
Not a silly question at all. This is not using vuex at all. Instead just using the `me` model. I did this for 2 reasons:
1. Simplicity of refactoring from the pug template -> vue component.
2. Ensures the User model interface is the same between CodeCombat and Ozaria.

Reply:
I guess the first part of my question is, it's generally useful to have `me` accessible in all templates, along with a few other things. With Vue, do we have a way to provide a general template context, [like we do in Backbone](https://github.com/codecombat/codecombat/blob/master/app/views/core/CocoView.coffee#L196), instead of needing to write explicit assignations for common context variables all the time?

Reply:
I think it's a really good idea. The main reason I haven't done that here is since the testing area is very large in this PR (as well as the similar Ozaria PR), and updating the global state of other vue components carries regression risks.

It's something worth discussing since it will reduce duplicate code.